Modules/Module_Processing/Module_Population/xASL_stat_GetROIstatistics.m

```diff
@@ -187,11 +187,12 @@
 namesROIlocal = x.S.NamesROI;
 
 if x.S.InputNativeSpace
-
-    
 	inputAtlasTmp = xASL_io_Nifti2Im(fullfile(x.dir.xASLDerivatives,x.SUBJECTS{1},listSessions{1},[x.S.InputAtlasNativeName '.nii']));
+	if x.S.bSubjectSpecificROI
+		inputAtlasTmp = round(inputAtlasTmp);

```

Comment:
Good point. Yes, atlases are binary. But while downsampling to native space, we can't use nearest neighbor - that would just not resample to the low-resolution properly. So that's why we smooth and interpolate.
Afterwards, rounding is perfectly valid to judge if a voxel is above of below 50% of the partial volume. This is the most correct approach. But I agree that we might do this thresholding elsewhere.

So I'll simply do this for atlases right in the resampling function. OK?

Reply:
That's fine, but note that we also allow MNI atlases to be a single 3D matrix, where each label is a different ROI.
E.g. 1 == caudate 2 == putamen etc.
Your approach is fine for a 4D matrix where each 3D matrix is a binary mask. But for a label image this won't work.

So, that means that we always need to convert a 3D label matrix to a 4D matrix with multiple binary 3D image volumes.

Additionally, nearest neighbor warping of course goes much faster than higher DoF. So if we could smooth and threshold in high res and interpolate with nearest neighbor that would be great, but you probably argue that that is incorrect :)

Most important is the label use case explained above, you can try /ExploreASL/External/Atlases4ROIs/LicenseLimited/Hammers.nii.gz

in native space as use case.

Reply:
We now transform Lesions not from T1, but from standard space. So we have the perilesional and contralateral ROIs created in standard space.

Moreover, this now works nicely with binary 4D masks. For 3 masks with multiple labels, we use nearest neighbor interpolation. We might want to change that in the future and do the transformation properly by binarizing the mask for each label.

Regarding your comment. Taking 3D multilabel and smoothing it and the interpolating with nearest neighbor isn't just incorrect, it won't work at all :) We would have to split it into a 4D binary. Which is rather easy and not even computationally difficult. I can easily implement that...

Reply:
Agree. New issue #1864